### PR TITLE
[refactor](load) move generate_delete_bitmap from memtable to beta rowset writer

### PR DESCRIPTION
**Pull request by kaijchen**
## Proposed changes

generate_delete_bitmap is not part of memtable logic, move it to beta rowset writer instead.
Also did some general cleanup in related code.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: be/src/olap/rowset/beta_rowset_writer.cpp [hunk 1]
```diff
@@ -141,6 +137,44 @@ Status BetaRowsetWriter::add_block(const vectorized::Block* block) {
     return _add_block(block, &_segment_writer);
 }
 
+Status BetaRowsetWriter::_generate_delete_bitmap(int32_t segment_id) {
+    SCOPED_RAW_TIMER(&_delete_bitmap_ns);
+    if (_context.tablet_schema->is_partial_update()) {
```
Comment (zhannngchen) [inline be/src/olap/rowset/beta_rowset_writer.cpp:142]: It is better to place this conditional statement after the conditional statement on L46.
Check MoW first, then partial update.

**Comment by kaijchen** [comment]
run buildall

File: be/src/olap/rowset/beta_rowset_writer.cpp (hunk 1 above)
Comment (kaijchen) [reply be/src/olap/rowset/beta_rowset_writer.cpp]: Thanks, updated.

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 37.56 seconds
 stream load tsv:          450 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          69.6 seconds inserted 10000000 Rows, about 143K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230630163550_clickbench_pr_170702.html

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**